### PR TITLE
Full 5x5 colorMatrix support

### PR DESCRIPTION
**Pull request by spaceribs**
Modified the colorMatrix fragment shader to support the full 5x5. An example of this fragment shader is implemented here:

http://codepen.io/spaceribs/pen/zxVLoe?editors=001

### CODE REVIEW

**Comment by spaceribs** [comment]
I'm still a little confused about the difference between the 5th column which represents the offset for each channel in some of the documentation I've looked at, and the 5th row which is also an offset for each channel here: https://msdn.microsoft.com/en-us/library/windows/desktop/ms533875(v=vs.85).aspx

Seems like a lot of inconsistency anyway, we could just simplify this to a 5x4?

**Comment by englercj** [comment]
@GoodBoyDigital Can you review this one? My matrix math needs some refreshing before I feel comfortable saying this is correct.

**Comment by GoodBoyDigital** [comment]
Hi @spaceribs. bit late to the party here - but yes we would accept a pull request for 5x4 . cheers man!

**Comment by spaceribs** [comment]
This should be good to go!

File: src/filters/color/ColorMatrixFilter.js [hunk 1]
```diff
@@ -113,10 +109,10 @@ ColorMatrixFilter.prototype._multiply = function (out, a, b) {
 /**
  * Create a Float32 Array and normalize the offset component to 0-1
  *
- * @param matrix {array} (mat 5x5)
- * @return m { array } (mat 5x5) with all values between 0-1
+ * @param matrix {Array.<number>} (mat 5x4)
```
Comment (englercj) [inline src/filters/color/ColorMatrixFilter.js]: Prefer the `number[]` syntax, easier to read! :)

File: src/filters/color/ColorMatrixFilter.js [hunk 2]
```diff
@@ -156,18 +150,19 @@ ColorMatrixFilter.prototype.brightness = function(b, multiply)
...
         scale, scale, scale, 0, 0,
-        0,0,0,1,0,
-        0,0,0,0,1
+        0, 0, 0, 1, 0
     ];
 
     this._loadMatrix(matrix, multiply);
 };
+//Americanized alias
+ColorMatrixFilter.prototype.grayscale = ColorMatrixFilter.prototype.greyscale;
```
Comment (englercj) [inline src/filters/color/ColorMatrixFilter.js]: Ha! I actually laughed when I saw this, good call!

File: .jshintrc [hunk 3]
```diff
@@ -33,11 +33,10 @@
     "trailing"      : true,     // Prohibit trailing whitespaces.
     "maxparams"     : 8,        // Prohibit having more than X number of params in a function.
     "maxdepth"      : 8,        // Prohibit nested blocks from going more than X levels deep.
-    "maxstatements" : false,    // Restrict the number of statements in a function.
-    "maxcomplexity" : false,    // Restrict the cyclomatic complexity of the code.
     "maxlen"        : 220,      // Require that all lines are 100 characters or less.
     "globals"       : {         // Register globals that are used in the code.
-        "performance": false    // Depends on polyfill, simplifies usage
+        "performance": false,   // Depends on polyfill, simplifies usage
+        "PIXI": false
```
Comment (englercj) [inline .jshintrc]: What was this added for?

Comment (spaceribs) [inline .jshintrc]: Isn't PIXI the libraries' global? I added it to cover any example files.

File: src/filters/color/ColorMatrixFilter.js (hunk 1 above)
Comment (spaceribs) [inline src/filters/color/ColorMatrixFilter.js]: cool, i'll switch that over now.

File: .jshintrc (hunk 3 above)
Comment (englercj) [inline .jshintrc]: Examples are in a completely different repo. This config is to lint the pixi source code which should not be using a global pixi object.

File: src/filters/color/ColorMatrixFilter.js (hunk 2 above)
Comment (spaceribs) [inline src/filters/color/ColorMatrixFilter.js]: I was half thinking I should now go through and make colour aliases...maybe a little extreme though

**Pull request merged**